lib/litani_report.py

```diff
@@ -319,22 +317,30 @@ def render(run, report_dir):
     with litani.atomic_write(report_dir / "index.html") as handle:
         print(page, file=handle)
 
-    with litani.atomic_write(report_dir / litani.RUN_FILE) as handle:
+    with litani.atomic_write(temporary_report_dir / litani.RUN_FILE) as handle:
         print(json.dumps(run, indent=2), file=handle)
 
     pipe_templ = env.get_template("pipeline.jinja.html")
     for pipe in run["pipelines"]:
         page = pipe_templ.render(run=run, pipe=pipe)
-        with litani.atomic_write(report_dir / pipe["url"]) as handle:
+        with litani.atomic_write(temporary_report_dir / pipe["url"]) as handle:
             print(page, file=handle)
 
+    temp_symlink_dir = report_dir.with_name(report_dir.name + "-tmp")
+    os.symlink(temporary_report_dir, temp_symlink_dir)
+    os.rename(temp_symlink_dir, report_dir)
+
 
 def render_artifact_indexes(artifact_dir):
     def dirs_needing_indexes():
         for root, dirs, fyles in os.walk(artifact_dir):
             if "index.html" not in fyles:
                 yield pathlib.Path(root), dirs, fyles
 
+    if artifact_dir.exists():
```

Comment:
I'm not sure that this check is necessary? `artifact_dir` is now underneath a temporary directory that you've just created. This condition seems impossible to me.